horizon/main.go

```diff
@@ -0,0 +1,113 @@
+package horizon
+
+import (
+	"encoding/json"
+	"errors"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"time"
+
+	"github.com/stellar/go-stellar-base/horizon/structs"
+)
+
+const TIMEOUT = 30 * time.Second
```

Comment:
I think having two timeouts is overcomplicating things.  IMO the best course is to preserve the golang http client default behavior (no timeout) and let consumers of the API set their own desired timeout.  We can simply add documentation that makes it known that submitting a transaction can take up to 10 seconds and to set your timeout accordingly if you need one.

Usually, I've seen this enabled by letting a consumer pass their own instance of `http.Client` into a constructor.  This has the added benefit of letting them perform additional customization, such as setting a proxy.